storage_service/storage_service/tests/test_password_validation.py

```diff
@@ -0,0 +1,44 @@
+# -*- coding: utf-8 -*-
+from __future__ import absolute_import
+
+import pytest
+
+from django.contrib.auth.password_validation import (
+    get_password_validators,
+    validate_password,
+)
+from django.core.exceptions import ValidationError
+
+
+@pytest.mark.parametrize(
+    "password,raises_exception",
+    [
+        # Passwords with only one type raise ValidationError.
+        ("abcdefghijk", True),
+        ("ABCDEFGHIJK", True),
+        ("12345678", True),
+        ("@#!_-=^$", True),
+        # Passwords with only two types raise ValidationError.
+        ("abcdEFGHIJK", True),
+        ("ABCDEF1234", True),
+        ("abcdef!$@", True),
+        # Passwords with three types pass validation.
+        ("abcdef123!@#", False),
+        ("ABCdef12345", False),
+        ("#@!abcdefGHIJK", False),
```

Comment:
It looks like accented characters work for upper case characters and I didn't realize regex was that agnostic, so that's cool! I wonder if you can perhaps add a few examples with different upper-case Unicode characters, e.g. some of `Á Â Ç Ë Ê Ñ Ö Ú Û` as a demonstration?

Reply:
Sure thing!

Reply:
Interesting - I decided to test this with commit https://github.com/artefactual/archivematica-storage-service/pull/553/commits/95feb94c5bc222e35faf3e6526e63e622a171b4d as I'm having trouble running tests in my dev environment at the moment (I think due to my failed Ubuntu upgrade before the break) and it looks like the tests are now failing. From a quick reading of the [output of the failed test run](https://github.com/artefactual/archivematica-storage-service/pull/553/checks?check_run_id=1659239104), it looks like the accented characters are not being counted as upper/lower-case letters by regex. Curious that it's different than what you saw.

Reply:
Ack, sorry @tw4l I was modifying that test when reviewing so I may have unhelpfully modified it to pass (I actually can't remember what I did, but I do remember I was testing something around the raises exception part of the code, so might have messed that up. Or got the intention of a test wrong.) 

I can verify that I see the same behavior - accented characters are not treated as upper/lower-case.

My personal feeling is that it would be good to recognize them as such, but I think that's an important question for you and the project team, and it largely out of scope for me in CR. So I'll trust your judgement on an approach.

Tests haven't been working for me in `tox` either with a storage service virtual environment. Because I think the other side of our testing woes is that they can fail inconsistently in Docker as a suite I've resorted to running them one by one, so I'm running: `docker-compose run --workdir /src --rm --no-deps --entrypoint py.test -e "DJANGO_SETTINGS_MODULE=storage_service.settings.test" archivematica-storage-service -p no:cacheprovider -p no:warnings -v storage_service/storage_service/tests/test_password_validation.py`. I haven't looked into the standalone environment.

Reply:
I agree with you that it's more correct to view accented alphabetical characters like their ASCII equivalents, so I've reworked the validator in commit https://github.com/artefactual/archivematica-storage-service/pull/553/commits/39599627341e4d1a3f852520e16b8dd5422c21b2 to achieve that. I looked at the [`regex` library on PyPI](https://pypi.org/project/regex/) per the top answer on [this SO post](https://stackoverflow.com/questions/36187349/python-regex-for-unicode-capitalized-words) but ultimately went with using Python's built-ins like `isupper()` and `is_digit()`, since those allow us to achieve the same result within bringing in an external library.

The validator is slightly more complex now, but I think still pretty minimal and effective. I'm curious to hear your thoughts on the approach taken.

Reply:
Also thanks for the pointer for now to run tests individually in Docker! That's really helpful

Reply:
Nice! I had started looking, but hadn't found a satisfying solution. I also leaned away from bringing in another library. This is great!